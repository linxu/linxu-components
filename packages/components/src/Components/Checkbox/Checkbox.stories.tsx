import React from "react";
import { Story } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import { getTheme } from "../../theme/getTheme";
import { ExomeCheckbox, CheckboxProps } from "./Checkbox";

export default {
  title: "Components/Checkbox",
  component: ExomeCheckbox,
};

export const Minimal: Story<CheckboxProps> = (props: CheckboxProps) => (
  <ThemeProvider theme={getTheme()}>
    <ExomeCheckbox {...props} />
  </ThemeProvider>
);

Minimal.args = {
  onChange: undefined,
  checked: true,
};
