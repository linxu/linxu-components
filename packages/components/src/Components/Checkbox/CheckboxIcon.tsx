import React from "react";
import { Svg } from "../../Icons/Svg/Svg";
import {Colors} from "../../theme/Colors";

interface Props {
  checked: boolean;
  color: keyof Colors;
}

export const CheckboxIcon = ({ checked, color }: Props) => (
  <Svg
    width="16"
    height="16"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
    color={color}
  >
    {checked && (
      <>
        <rect x="2" y="2" width="12" height="12" fill="white" />
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM3.66602 8.66803L6.5436 11.666L12.3327 5.66298L11.3758 4.66602L6.5436 9.67209L4.62294 7.67106L3.66602 8.66803Z"
        />
      </>
    )}
    {!checked && (
      <>
        <rect
          x="1.33398"
          y="1.33398"
          width="13.3333"
          height="13.3333"
          fill="white"
        />
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM2.99935 1.99935C2.44706 1.99935 1.99935 2.44706 1.99935 2.99935V12.9993C1.99935 13.5516 2.44706 13.9993 2.99935 13.9993H12.9993C13.5516 13.9993 13.9993 13.5516 13.9993 12.9993V2.99935C13.9993 2.44706 13.5516 1.99935 12.9993 1.99935H2.99935Z"
        />
      </>
    )}
  </Svg>
);
