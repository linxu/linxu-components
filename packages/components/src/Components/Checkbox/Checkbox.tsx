import React from "react";
import styled from "styled-components";
import MUICheckbox from "@material-ui/core/Checkbox";
import { CheckboxIcon } from "./CheckboxIcon";

export interface CheckboxProps {
  // If true, the component is checked */
  checked: boolean;
  // Callback fired when the state is changed */
  onChange: () => void;
}

// ScCheckbox Component <br>
// Other Props Reference https://material-ui.com/zh/api/checkbox/
export const Checkbox = styled(MUICheckbox).attrs<CheckboxProps>(() => ({
  icon: <CheckboxIcon checked={false} color="secondary" />,
  checkedIcon: <CheckboxIcon checked color="success" />,
}))``;

export const ExomeCheckbox = Checkbox;
