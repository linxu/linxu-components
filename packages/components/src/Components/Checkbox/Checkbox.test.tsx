import React from "react";
import { render } from "@testing-library/react";

import { Minimal } from "./Checkbox.stories";
import { CheckboxProps } from "./Checkbox";

it("renders the badge with color", async () => {
  const tree = render(
    <Minimal {...(Minimal.args as CheckboxProps)} checked={false} />
  );
  expect(tree).toMatchSnapshot();
});
