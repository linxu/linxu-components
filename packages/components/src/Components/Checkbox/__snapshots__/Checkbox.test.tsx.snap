// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the badge with color 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary sc-gsTEea MuiIconButton-colorSecondary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-4"
            data-indeterminate="false"
            type="checkbox"
            value=""
          />
          <svg
            class="sc-bdfBQB kleLNI"
            color="secondary"
            fill="#21222F"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="white"
              height="13.3333"
              width="13.3333"
              x="1.33398"
              y="1.33398"
            />
            <path
              clip-rule="evenodd"
              d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM2.99935 1.99935C2.44706 1.99935 1.99935 2.44706 1.99935 2.99935V12.9993C1.99935 13.5516 2.44706 13.9993 2.99935 13.9993H12.9993C13.5516 13.9993 13.9993 13.5516 13.9993 12.9993V2.99935C13.9993 2.44706 13.5516 1.99935 12.9993 1.99935H2.99935Z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary sc-gsTEea MuiIconButton-colorSecondary"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-4"
          data-indeterminate="false"
          type="checkbox"
          value=""
        />
        <svg
          class="sc-bdfBQB kleLNI"
          color="secondary"
          fill="#21222F"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="white"
            height="13.3333"
            width="13.3333"
            x="1.33398"
            y="1.33398"
          />
          <path
            clip-rule="evenodd"
            d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM2.99935 1.99935C2.44706 1.99935 1.99935 2.44706 1.99935 2.99935V12.9993C1.99935 13.5516 2.44706 13.9993 2.99935 13.9993H12.9993C13.5516 13.9993 13.9993 13.5516 13.9993 12.9993V2.99935C13.9993 2.44706 13.5516 1.99935 12.9993 1.99935H2.99935Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
